### ISSUE TITLE
RedirectController parameters pop

### ISSUE THREAD

**Issue by clementmas** [laravel/framework#30639]
- Laravel Version: 6.5.2
- PHP Version: 7.2.16

### Description:
Since I upgraded to Laravel 6, the `Route::redirect` helper is throwing the error:
> Argument 2 passed to Symfony\Component\HttpFoundation\RedirectResponse::__construct() must be of the type integer, null given

### Steps To Reproduce:
On a fresh Laravel install, add these routes:
```php
Route::domain('{username}.laravel.test')->group(function() {
    Route::get('/', function () {
        return view('welcome');
    });

    Route::redirect('/old', '/');
});
```

And add this binding in the `RouteServiceProvider`:
```php
public function boot()
{
    parent::boot();

    Route::bind('username', function ($value, $route) {
        $route->forgetParameter('username');
    });
}
```
Now visiting `test.laravel.test/old` will throw an error.

### Source of the bug:
The `status` and `destination` parameters of the [RedirectController](https://github.com/laravel/framework/blob/6.x/src/Illuminate/Routing/RedirectController.php#L22) should not be retrieved using `pop()`:
```php
class RedirectController extends Controller
{
    public function __invoke(Request $request, UrlGenerator $url)
    {
        $parameters = collect($request->route()->parameters());

        $status = $parameters->pop();

        $destination = $parameters->pop();

        // [...]
    }
}
```
This is the dump of the `$parameters` with the `username` binding:
```
Collection {#228 ▼
  #items: array:3 [▼
    "destination" => "/"
    "status" => 302
    "username" => null
  ]
}
```
So `$status = null` and `$destination = 302`.

Now I know this is probably not a common scenario but `pop()` should be replaced with something like `$parameters['status']` and `$parameters['destination']` to avoid any bug.

Would you accept a PR to change this? Or am I missing something?

**Comment by driesvints** [comment]
Hey there,

Can you first please try one of the support channels below? If you can actually identify this as a bug, feel free to report back and I'll gladly help you out and re-open this issue.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

Thanks!

**Comment by clementmas** [comment]
Hi Dries, I thought I already identified the source of the bug in my post. What exactly do you need to re-open this issue?
Unless I'm not supposed to use `forgetParameter` with subdomain parameters, then it is a bug.
Thanks

**Comment by driesvints** [comment]
It doesn't seems likely that you need to call `forgetParameter` since it's an undocumented method. Can you please try a support channel?

**Comment by clementmas** [comment]
Ok, I posted the issue on the [Laracasts Forum](https://laracasts.com/discuss/channels/laravel/routeredirect-helper-bug).

As @snapey is saying, since the `forgetParameter` method is public and described in the API, then there's no reason it should lead to a bug.

That's why I suggested a PR to fix the code in `RedirectController`.
Either by adding `->parametersWithoutNulls()` to the first line as @ismaile suggested or by replacing the parameters popping with a cleaner solution.
